psi4/src/psi4/scfgrad/response.cc

```diff
@@ -1052,16 +1052,18 @@ std::shared_ptr<Matrix> SCFGrad::rhf_hessian_response()
         } // End if density fitted
     }
 
+    size_t mem = 0.9 * memory_ / 8L;
+    size_t per_A = 3L * nso * nso + 1L * nocc * nso;
+    size_t max_A = (mem / 2L) / per_A;
+    max_A = (max_A > 3 * natom ? 3 * natom : max_A);
+    
     std::shared_ptr<JK> jk;
     if (options_.get_str("SCF_TYPE") == "DF"){
-        jk = JK::build_JK(basisset_, get_basisset("DF_BASIS_SCF"), options_);
+        jk = JK::build_JK(basisset_, get_basisset("DF_BASIS_SCF"), options_, false, mem);
```

Comment:
Why go into an `if` statement here? This signature should be valid for all SCF_TYPE's?

Reply:
Yes, the signature is valid for all `SCF_TYPE`'s, I was merely using the `if` statements that were already in place.  This check occurs in almost every location of `build_JK()`